### PR TITLE
[ONNX] Move tests in test_onnx_export.py to test_pytorch_onnx_no_runtime.py

### LINKED ISSUE #78308: Move test/onnx/test_onnx_export.py into test_pytorch_onnx_no_runtime.py
To integrate no runtime test cases into one file, test/onnx/test_onnx_export.py (added #74759) should be moved into test_pytorch_onnx_no_runtime.py (added #73284)

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #78308
This should be merged after
- #78116

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 1]
```diff
@@ -2,22 +2,77 @@
...
-from typing import Optional, Type
+from typing import Dict, Optional, Type, Callable, Iterable, Tuple, Union
 
 import onnx
 
 import torch
 from torch import Tensor
-from torch.onnx import symbolic_helper
+from torch.onnx import OperatorExportTypes, symbolic_helper, utils, symbolic_registry
+from torch.onnx._globals import GLOBALS
```
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_no_runtime.py:16]: Good time to clean up import lines that import functions and classes

**Review by justinchuby — APPROVED** [review]
LGTM🚀

**Comment by justinchuby** [comment]
I would update the title to a more descriptive one

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 2]
```diff
@@ -107,8 +155,427 @@ def forward(self, y: Optional[Tensor]) -> Optional[Tensor]:
...
+        def cast_device_cpu_string(src: torch.Tensor) -> torch.Tensor:
+            return src.to(device="cpu")
+
+        self._helper_test_to_(cast_device_cpu_string)
+
 
+common_utils.instantiate_parametrized_tests(TestOptionalOutput)
 
 if __name__ == "__main__":
     unittest.main()
```
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_no_runtime.py:581]: Should we use `common_utils.run_tests()`?

**Comment by justinchuby** [comment]
ONNX tests are failing because the test script specifically looks for the test_onnx_export file. I would coordinate with @BowenBao on #78200

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 3]
```diff
@@ -107,8 +155,427 @@ def forward(self, y: Optional[Tensor]) -> Optional[Tensor]:
         )
 
 
-instantiate_parametrized_tests(TestOptionalOutput)
+class TestONNXExport(common_utils.TestCase):
```
Comment (AllenTiTaiWang) [inline test/onnx/test_pytorch_onnx_no_runtime.py:158]: In this case, common_utils.Testcase is used as assertAtenOp() is its member function, and we use that to check Aten fallback.

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 4]
```diff
@@ -2,19 +2,70 @@
...
+            opset_version=opset_version,
+        )
+
+    # Validate ONNX graph before returning it
+    onnx_model = onnx.load_from_string(f.getvalue())
+    onnx.checker.check_model(onnx_model)
+    return onnx_model
+
+
+@common_utils.instantiate_parametrized_tests
```
Comment (AllenTiTaiWang) [inline test/onnx/test_pytorch_onnx_no_runtime.py:67]: Using parametrized decorator @justinchuby

**Review by justinchuby — APPROVED** [review]
LGTM🚀

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 5]
```diff
@@ -453,8 +503,78 @@ def forward(self, input):
...
+        def cast_device_cpu_string(src: torch.Tensor) -> torch.Tensor:
+            return src.to(device="cpu")
+
+        self._helper_test_to_(cast_device_cpu_string)
 
-common_utils.instantiate_parametrized_tests(TestOptionalOutput)
 
 if __name__ == "__main__":
-    unittest.main()
+    common_utils.run_tests()
```
Comment (AllenTiTaiWang) [inline test/onnx/test_pytorch_onnx_no_runtime.py:580]: common_utils.run_tests() has more functionality than unittest.main() and also more informative error message if anything went off.

**Comment by justinchuby** [comment]
cc @garymm for approval

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 6]
```diff
@@ -2,19 +2,70 @@
...
+
+    # Validate ONNX graph before returning it
+    onnx_model = onnx.load_from_string(f.getvalue())
+    onnx.checker.check_model(onnx_model)
+    return onnx_model
+
+
+@common_utils.instantiate_parametrized_tests
+class TestOptionalOutput(common_utils.TestCase):
     # TODO: Move these tests to test_pytorch_onnx_onnxruntime once
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_no_runtime.py:69]: can you please update this to reference a GitHub issue that tracks this? Can be in ORT or PyTorch

**Review by garymm — APPROVED** [review]

File: test/onnx/test_pytorch_onnx_no_runtime.py (hunk 6 above)
Comment (AllenTiTaiWang) [reply test/onnx/test_pytorch_onnx_no_runtime.py:69]: https://github.com/microsoft/onnxruntime/issues/11717

**Comment by justinchuby** [comment]
You may need to remove `test_onnx_export.py` from https://github.com/pytorch/pytorch/blob/eda8f69aee50807734b50ff8bf7b93b2e2bba893/scripts/onnx/test.sh#L73

**Pull request closed without merging**